std-bits/base/src/main/java/org/enso/base/Polyglot_Utils.java

```diff
@@ -0,0 +1,42 @@
+package org.enso.base;
+
+import org.graalvm.polyglot.Value;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+public class Polyglot_Utils {
+  public static Object convertPolyglotValue(Value item) {
+    if (item.isDate()) {
+      LocalDate d = item.asDate();
+      if (item.isTime()) {
+        LocalDateTime dtime = d.atTime(item.asTime());
+        if (item.isTimeZone()) {
+          return dtime.atZone(item.asTimeZone());
+        } else {
+          return dtime;
+        }
+      } else {
+        return d;
+      }
+    } else if (item.isTime()) {
+      return item.asTime();
+    }
+
+    return item.as(Object.class);
+  }
+
+  /** A helper functions for situations where we cannot use the Value conversion directly.
+   *
+   * Mostly happens due to the issue: TODO
```

Comment:
Missing TODO link?

Reply:
I was still working on the repro, so the link was just not ready :)
Probably should have marked the PR as draft in the meantime, sorry.